api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDaoImpl.java

```diff
@@ -418,13 +418,14 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		return Optional.empty();
 	}
 	
-	protected Optional<Criterion> handleEncounterReference(@NotNull String encounterAlias,
-	        ReferenceParam encounterReference) {
-		if (encounterReference == null || encounterReference.getIdPart() == null) {
-			return Optional.empty();
+	protected void handleEncounterReference(Criteria criteria, ReferenceAndListParam encounterReference) {
+		if (encounterReference != null) {
+			criteria.createAlias("encounter", "e");
+			handleAndListParam(encounterReference, token -> {
+				criteria.add(ilike("e.uuid", token.getIdPart()));
+				return Optional.empty();
+			});
```

Comment:
I would prefer that instead of creating the alias here, we just pass in what the alias is called as it was before.